### ISSUE TITLE
Dependency conflicts of numpy versions

### ISSUE THREAD

**Issue by V-Enzo** [thuml/TrajWorld#2]
Hi, @operator22th 

Thank you for sharing your excellent work!

I am trying to run the code as readme, but I met the conflicts of Numpy versions when running `conda env create -f environment.yaml`

The error information is like

```
The conflict is caused by,
    The user requested numpy==1.26.1
    arch 7.2.0 depends on numpy>=1.22.3
    chex 0.1.88 depends on numpy>=1.24.1
    contourpy 1.2.1 depends on numpy>=1.20
    d4rl 1.1 depends on numpy
    distrax 0.1.5 depends on numpy>=1.23.0
    dm-control 1.0.21 depends on numpy>=1.9.0
    dm-env 1.6 depends on numpy
    flax 0.8.5 depends on numpy>=1.22
    gym 0.23.1 depends on numpy>=1.18.0
    gymnasium 1.0.0a2 depends on numpy>=1.21.0
    h5py 3.11.0 depends on numpy>=1.17.3
    imageio 2.34.2 depends on numpy
    jax 0.4.31 depends on numpy>=1.24
    jaxlib 0.4.31 depends on numpy>=1.24
    keras 3.6.0 depends on numpy
    labmaze 1.0.6 depends on numpy>=1.8.0
    mani-skill2 0.5.3 depends on numpy<1.24
To fix this you could try to:
1. loosen the range of package versions you've specified
2. remove package versions to allow pip attempt to solve the dependency conflict
```
I tried to modify the version for compatibility, but met a segmentation fault in fine-tuning with
```
python scripts/training/train_trajworld.py 
env_name=hopper-medium-replay-v2 
log_root_dir=log_model_new 
trm_epoch_steps=5000 
dynamics_max_epochs_since_update=300
dynamics_max_epochs=50 
seed=183 
train_model_only=true 
exp_name=trajworld_scratch 
trm_lr=1e-5 
n_blocks=6
```
Would you double-check the environment setting?

Thank you very much!

**Comment by operator22th** [comment]
Hi, thanks for pointing that out. Could you try forcibly installing numpy==1.26.1, or alternatively remove the mani-skill2 requirement? I believe mani-skill2 isn’t actually needed in this codebase.

**Comment by operator22th** [comment]
Hi, it’s the same file under a different name. I just tried the command and was able to run the fine-tuning process from the README successfully by simply replacing `"mergeall_pt/model/trm_dynamics990000.pkl"` with `"pt_model/trajworld_pt/trajworld_pt.pkl"`. I’m not sure what’s going wrong on your side.

**Comment by operator22th** [comment]
Thanks for bringing this up. I’ve updated `environment.yaml` and the `README`.

**Comment by operator22th** [comment]
As for the core dump issue, I haven’t encountered that before. Could you provide some context?

**Comment by V-Enzo** [comment]
Sure. This is the error I get after finetuning 5000 steps `python scripts/training/train_trajworld.py env_name=hopper-medium-replay-v2 log_root_dir=log_model_new trm_epoch_steps=5000 dynamics_max_epochs_since_update=300 dynamics_max_epochs=50 seed=183 train_model_only=true exp_name=trajworld_ft trm_lr=1e-5 load_pt_dynamics_path="pt_model/trajworld_pt/trajworld_pt.pkl" n_blocks=6`

```
2025-08-12 09:39:14.056260: W external/xla/xla/service/gpu/nvptx_compiler.cc:836] The NVIDIA driver's CUDA version is 12.4 which is older than the PTX compiler version (12.6.20). Because the driver is older than the PTX compiler version, XLA is disabling parallel compilation, which may slow down compilation. You should update your NVIDIA driver or use the NVIDIA-provided CUDA forward compatibility packages.
holdout_indices [115884 115885 115886 115887 115888 115889 115890 115891 115892 115893]
Training dynamics:
training dynamics: 100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 5000/5000 [03:54<00:00, 21.34it/s, loss=0.4959117]
[1]    25309 segmentation fault (core dumped)  python scripts/training/train_trajworld.py env_name=hopper-medium-replay-v2
```
I am using the CUDA 12.4 (older than the PTX compiler version) on RTX 4090 and I think probably the following config in 'scripts/training/train_trajworld.py' caused the problem
```
os.environ["XLA_FLAGS"] = "--xla_gpu_triton_gemm_any=True "
os.environ["XLA_PYTHON_CLIENT_PREALLOCATE"] = "false"
```

I modified the config to
```
os.environ["XLA_FLAGS"] = "--xla_gpu_triton_gemm_any=False --xla_gpu_force_compilation_parallelism=1"
os.environ["XLA_PYTHON_CLIENT_PREALLOCATE"] = "false"
os.environ["XLA_PYTHON_CLIENT_MEM_FRACTION"] = "0.8"
```
which can execute successfully.

Any advice is appreciated. Thanks!

**Comment by operator22th** [comment]
Great to hear the problem’s been fixed! I’ll close the issue now.
